packages/tool-cache/src/tool-cache.ts

```diff
@@ -454,6 +455,42 @@ export function findAllVersions(toolName: string, arch?: string): string[] {
   return versions
 }
 
+// versions-manifest
+//
+// typical pattern of a setup-* action that supports JIT would be:
+// 1. resolve semver against local cache
+//
+// 2. if no match, download
+//   a. query versions manifest to match
+//   b. if no match, fall back to source if exists (tool distribution)
+//   c. with download url, download, install and preprent path
+
+export type IToolRelease = mm.IToolRelease
+export type IToolReleaseFile = mm.IToolReleaseFile
+
+export async function getManifestFromUrl(url: string): Promise<IToolRelease[]> {
+  const http: httpm.HttpClient = new httpm.HttpClient('tool-cache')
+
+  //let versions: IToolRelease[] =
+  return (await http.getJson<IToolRelease[]>(url)).result || []
+  //return
+}
+
+export async function findFromManifest(
```

Comment:
As I understand, currently, we should use toolkit the following way:
```
const manifest = toolkit.getManifestFromUrl("<url>");
const release = toolkit.findFromManifest(versionSpec, stable, manifest);
```

I think it will be better to pass manifestUrl to `findFromManifest` and invoke `getManifestFromUrl` there:
```
export async function findFromManifest(
  versionSpec: string,
  stable: boolean,
  manifestUrl: string
): Promise<IToolRelease | undefined> {
  const manifest = getManifestFromUrl(manifestUrl);
  const match: mm.IToolRelease | undefined = await mm._findMatch(
    versionSpec,
    stable,
    manifest
  )

  return match
}

```
So it will be easier to use

Reply:
Good suggestion.  I will make this change.

Reply:
Actually, I remembered why I split.  While the common case may be to get the manifest from an url, I didn't want to dictate that or allow it to come from other places.   Separating the retrieval of the manifest from the querying, allows that flexibility and two lines is still not bad :)